src/ApiClient.ts

```diff
@@ -237,6 +240,10 @@ export default class ApiClient {
           if (new Date().getTime() - startTime < 1000) {
             await sleep(1000)
           }
+
+          if (onConnectionLost) {
+            onConnectionLost()
+          }
```

Comment:
super nit for these 2 calls: WDYT of doing `onConnectionLost?.()` instead using the `if` statement? definitely not a blocker!